software/messaging/src/main/java/brooklyn/entity/messaging/rabbit/RabbitSshDriver.java

```diff
@@ -72,8 +80,15 @@ public void preInstall() {
     public void install() {
         List<String> urls = resolver.getTargets();
         String saveAs = resolver.getFilename();
+        String osMajorVersion = Strings.nullToEmpty(getMachine().getOsDetails().getVersion());
+        osMajorVersion = osMajorVersion.indexOf(".") > 0 ? osMajorVersion.substring(0, osMajorVersion.indexOf('.')) : osMajorVersion;
+        String epelVersion = Strings.nullToEmpty(CENTOS_VERSION_TO_EPEL_VERSION.get(osMajorVersion));
+        String osArchitecture = Strings.nullToEmpty(getMachine().getOsDetails().getArch());
```

Comment:
Is this behaviour desirable? NPEs are guarded against but it could generate URLs like `http://download.fedoraproject.org/pub/epel///epel-release-.noarch.rpm`. Should we pick a default if we can't detect versions and architectures? Or it could even just fail.